preprocessors/openstreetmap/osm_service.py

```diff
@@ -492,9 +492,63 @@ def validate_new_node_coordinates(lat2, lon2, bbox_coordinates):
         return True
 
 
+def compute_segment_slope(x1, y1, x2, y2):
+    if (x2 - x1) != 0:
+        segment_slope = (y2 - y1) / (x2 - x1)
+        return segment_slope
+    return sys.maxsize
+
+
+def compute_street_segments_angle(segment1, segment2, intersecting_point):
+    intersecting_point_index1 = segment1.index(intersecting_point[0])
+    intersecting_point_index2 = segment2.index(intersecting_point[0])
+
+    x0 = intersecting_point[0]["lon"]
+    y0 = intersecting_point[0]["lat"]
+
+    if intersecting_point_index1 == 0:
+        x1 = segment1[intersecting_point_index1 + 1]["lon"]
+        y1 = segment1[intersecting_point_index1 + 1]["lat"]
+
+    if intersecting_point_index2 == 0:
+        x2 = segment2[intersecting_point_index2 + 1]["lon"]
+        y2 = segment2[intersecting_point_index2 + 1]["lat"]
+
+    if intersecting_point_index1 > 0 and intersecting_point_index1 < len(
+            segment1) - 1:
+        x1 = segment1[intersecting_point_index1 - 1]["lon"]
+        y1 = segment1[intersecting_point_index1 - 1]["lat"]
+
+    if intersecting_point_index2 > 0 and intersecting_point_index2 < len(
+            segment2) - 1:
+        x2 = segment2[intersecting_point_index2 - 1]["lon"]
+        y2 = segment2[intersecting_point_index2 - 1]["lat"]
+
+    if intersecting_point_index1 == len(segment1) - 1:
+        x1 = segment1[intersecting_point_index1 - 1]["lon"]
+        y1 = segment1[intersecting_point_index1 - 1]["lat"]
+
+    if intersecting_point_index2 == len(segment2) - 1:
+        x2 = segment2[intersecting_point_index2 - 1]["lon"]
+        y2 = segment2[intersecting_point_index2 - 1]["lat"]
+
+    segment1_slope = compute_segment_slope(x1, y1, x0, y0)
+    segment2_slope = compute_segment_slope(x0, y0, x2, y2)
+    angle = abs((segment2_slope - segment1_slope)) / \
+        (1 + (segment1_slope * segment2_slope))
+    theta = math.atan(angle) * 180 / math.pi  # theta in degrees
+    return theta
+
+
 def compare_street(street1, street2):  # Compare two streets
-    intersecting_points = [x for x in street1 if x in street2]
-    return intersecting_points
+    intersecting_point = [x for x in street1 if x in street2]
+    if len(intersecting_point):
+        theta = compute_street_segments_angle(
```

Comment:
Maybe I should have created a separate request. This addresses the two segments of a street being identified as intersecting when they are not. The discussion is still ongoing as I hope to meet with Vennisa today because choosing an optimal angle of intersection that would correctly identify and separate them seems not to be feasible in all cases, going by the way OpenStreetMap currently has its data structured. I'm not sure, but we might eventually have to look for other options.

Reply:
Yes that is an entirely separate issue and should not be included in this PR, especially since it needed additional discussion as of Friday.